ksql-common/src/main/java/io/confluent/ksql/util/KsqlConfig.java

```diff
@@ -196,10 +196,22 @@
           + "whether the Kafka cluster supports the required API, and enables the validator if "
           + "it does.";
 
-  public static final String KSQL_PULL_QUERIES_ENABLE_CONFIG = "ksql.pull.queries.enable";
-  public static final String KSQL_PULL_QUERIES_ENABLE_DOC =
+  public static final String KSQL_QUERY_PULL_ENABLE_CONFIG = "ksql.query.pull.enable";
+  public static final String KSQL_QUERY_PULL_ENABLE_DOC =
       "Config to enable or disable transient pull queries on a specific KSQL server.";
-  public static final boolean KSQL_PULL_QUERIES_ENABLE_DEFAULT = true;
+  public static final boolean KSQL_QUERY_PULL_ENABLE_DEFAULT = true;
+
+  public static final String KSQL_QUERY_PULL_ROUTING_TIMEOUT_MS_CONFIG =
+      "ksql.query.pull.routing.timeout.ms";
+  public static final Long KSQL_QUERY_PULL_ROUTING_TIMEOUT_MS_DEFAULT = 30000L;
+  public static final String KSQL_QUERY_PULL_ROUTING_TIMEOUT_MS_DOC = "Timeout in milliseconds "
+      + "when waiting for the lookup of the owner of a row key";
+
+  public static final String KSQL_STREAMSTORE_REBALANCING_TIMEOUT_MS_CONFIG =
+      "ksql.streamsstore.rebalancing.timeout.ms";
+  public static final Long KSQL_STREAMSTORE_REBALANCING_TIMEOUT_MS_DEFAULT = 10000L;
+  public static final String KSQL_STREAMSTORE_REBALANCING_TIMEOUT_MS_DOC = "Timeout in "
+      + "milliseconds when waiting for rebalancing of the stream store during a write";
```

Comment:
Please change "during a write"  => "during a pull query". its a read.. Another thing is, do we also want to scope this config to pull queries, since that's where this is used.. `ksql.query.pull.streamsstore.rebalancing.timeout.ms` ? may be its too long. I leave it to your call.

Reply:
Ah, the method call was a little confusing.  I assumed that since it was called "store", it was a write, but it's actually fetching a store.

Changed name to the one you recommend.